src/reducer/filterReducer.jsx

```diff
@@ -0,0 +1,39 @@
+import { initialState } from "../helper";
+
+const filterReducer=(state,action)=>{
+    switch(action.type){
+        case "FILTER_BY_PRICE":
+            return {...state,pricerange:action.payload}
+
+        case "FILTER_BY_BRANDS":
+            return {...state,listofbrands:state.listofbrands.includes(action.payload)?
+                state.listofbrands.filter(ele=>ele!==action.payload):
+                [...state.listofbrands,action.payload]}
+
+        case "1_STAR_AND_ABOVE":
+            return {...state,rating:action.payload}
+
+        case "2_STARS_AND_ABOVE": 
+            return {...state,rating:action.payload}
+
+        case "3_STARS_AND_ABOVE":
+            return {...state,rating:action.payload}
+
```

Comment:
Instead of putting 5 cases for the same filter, you can do it within one case and compare according to the payload you give

Reply:
I have made the suggested changes.